cpp/CMakeLists.txt

```diff
@@ -804,7 +809,7 @@ target_link_libraries(
   cudf
   PUBLIC ${ARROW_LIBRARIES} CCCL::CCCL rmm::rmm
   PRIVATE $<BUILD_LOCAL_INTERFACE:nvtx3::nvtx3-cpp> cuco::cuco ZLIB::ZLIB nvcomp::nvcomp
-          kvikio::kvikio $<TARGET_NAME_IF_EXISTS:cuFile_interface> nanoarrow
+          kvikio::kvikio $<TARGET_NAME_IF_EXISTS:cuFile_interface> nanoarrow BS_thread_pool
```

Comment:
It looks like benchmark builds are failing. Since this dependency is PRIVATE, it is not exported to the benchmarks. You probably need to add this to the benchmarks CMakeList.txt as well.